src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -334,6 +336,17 @@ func (r *ClusterReconciler) handlePodFinalizer(
 	if err != nil {
 		return fmt.Errorf("unable to fetch PodList: %w", err)
 	}
+
+	var decommissionOnDelete bool
+	decommissionOnDeleteVal, ok := rp.Annotations[DecommissionOnDeleteAnnotation]
```

Comment:
Can you show me the effect of this being true? I checked and this seems like a new boolean, and found non, so how does decommission get triggered now?